lib/bibliothecary/parsers/go.rb

```diff
@@ -7,8 +7,12 @@ class Go
       include Bibliothecary::Analyser
 
       GPM_REGEXP = /^(.+)\s+(.+)$/
-      GOMOD_REGEX = /^(require\s+)?(.+)\s+(.+)$/
-      GOMOD_IGNORABLE_REGEX = /^(\/\/|module\s|go\s|exclude\s|replace\s|require\s+\(|\))/m
+      GOMOD_REPLACEMENT_SEPARATOR_REGEX = / => /
+      GOMOD_DEP_REGEX = /(?<name>\S+)\s?(?<requirement>[^ =>]+)?/
+      GOMOD_SINGLELINE_DEP_REGEX = /^(?<category>require|exclude|replace|retract)\s+#{GOMOD_DEP_REGEX}.*$/
+      GOMOD_MULTILINE_START_REGEX = /^(?<category>require|exclude|replace|retract)\s+\(/
+      GOMOD_MULTILINE_END_REGEX = /^\)/
+      GOMOD_MULTILINE_DEP_REGEX = /^#{GOMOD_DEP_REGEX}.*$/
```

Comment:
go.mod directives can be either singleline (`require foo v1.0.0`) or multiline (`require (\n foo v1.0.0\n)`) so this PR adds more intelligent parsing for both cases.